### DISCUSSION TITLE
Dbeaver using only 1 processor core,

### DISCUSSION THREAD

**Discussion by dyazbek21** [dbeaver/dbeaver#35047] [General]
When exporting or using something high demand, dbeaver uses 100% on 1 core and the others 7 cores 1% to 5% use
Dbeaver have some configuration or option to enable all cores?

**Comment by ask9** [comment]
I fully support you initiative.
You may enumerate the list of heavy long-running tasks you want to make use of multiple cores.
Also consider, that it may be impossible to parallelize some kinds of tasks.

**Reply by dyazbek21** [reply]
Normally an sql query with some joins and each table have around 300.000+ rows and the joins give the result with more than 700.000 rows
Is not a mega huge database, but I saw the processing use and take a loooong time to execute using 1 core.

**Reply by ask9** [reply]
Have you ensured that the database engine itself configured properly to make use of all cpu cores?
And the query you are executing can use parallel plan execution?
Does it work faster when you execute the query in sql editor and slowly on export?

I don't know a lot, but, for example, if we talk about postgresql, here is the starting point about multiple cores:
https://www.postgresql.org/docs/current/parallel-query.html

It may be not a DBeaver's issue but configuration of your environment or the particular query.
Without a detailed information on how to reproduce it there is no way to fix it or help you somehow.
